js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorer.tsx

```diff
@@ -147,7 +148,13 @@ const AssetGraphExplorerWithData: React.FC<WithDataProps> = ({
   allAssetKeys,
 }) => {
   const findAssetLocation = useFindAssetLocation();
-  const {layout, loading, async} = useAssetLayout(assetGraphData);
+  const pathname = useHistory().location.pathname;
+  const {layout, loading, async} = useAssetLayout(
+    assetGraphData,
+    // Use the pathname as part of the key so that different deployments don't invalidate each other's cached layout
+    `explorer:${pathname}`,
```

Comment:
Hmm I think we may actually want this cache to be shared to limit the amount of data that this feature can keep in local storage a bit more.

Reply:
The problem with sharing it is that different deployments (PROD/Staging, etc.) will overwrite each others cache data.